the-blue-alliance-ios/View Controllers/Teams/TeamSummaryTableViewController.swift

```diff
@@ -1,16 +1,320 @@
+            summaryRows.append(TeamSummaryRow.record)
+            summaryValues.append(record)
+        }
+        
+        // Alliance
+        if let allianceStatusString = eventStatus?.allianceStatus {
+            summaryRows.append(TeamSummaryRow.alliance)
+            summaryValues.append(allianceStatusString)
+        }
+        
+        // Team Status
+        if let overallStatusString = eventStatus?.overallStatus {
+            summaryRows.append(TeamSummaryRow.status)
+            summaryValues.append(overallStatusString)
+        }
+        
+        // Breakdown
+        if let breakdown = eventStatus?.qual?.ranking?.infoString {
+            summaryRows.append(TeamSummaryRow.breakdown)
+            summaryValues.append(breakdown)
+        }
+
+        // Only show next/last match if the event is happening now
+        // TODO: Change this back Zach
+        if !event.isHappeningNow {
+            for (type, key) in [(TeamSummaryRow.nextMatch, eventStatus?.nextMatchKey), (TeamSummaryRow.lastMatch, eventStatus?.lastMatchKey)] {
+                guard let key = key else {
+                    continue
+                }
+                summaryRows.append(type)
+                let match = Match.findOrFetch(in: persistentContainer.viewContext, matching: NSPredicate(format: "%K == %@", #keyPath(Match.key), key))
+                if let match = match {
+                    summaryValues.append(match)
+                } else {
+                    summaryValues.append(key)
+                }
+            }
+        }
+        
+        DispatchQueue.main.async { [weak self] in
+            self?.tableView.reloadData()
+        }
+    }
+    
+    // MARK: - Refresh
     
     override func refresh() {
+        removeNoDataView()
         
+        var request: URLSessionDataTask?
+        request = TBAKit.sharedKit.fetchTeamStatus(key: team.key!, eventKey: event.key!, completion: { (modelStatus, error) in
+            if let error = error {
+                self.showErrorAlert(with: "Unable to refresh event - \(error.localizedDescription)")
+            }
+            
+            self.persistentContainer?.performBackgroundTask({ (backgroundContext) in
+                let backgroundTeam = backgroundContext.object(with: self.team.objectID) as! Team
+                let backgroundEvent = backgroundContext.object(with: self.event.objectID) as! Event
+
+                if let modelStatus = modelStatus {
+                    _ = EventStatus.insert(with: modelStatus, team: backgroundTeam, event: backgroundEvent, in: backgroundContext)
+                }
+                
+                if !backgroundContext.saveOrRollback() {
+                    self.showErrorAlert(with: "Unable to refresh event - database error")
+                }
+                self.removeRequest(request: request!)
+            })
+        })
+        addRequest(request: request!)
```

Comment:
Kickoff `/team/{team_key}/event/{event_key}/awards` and add it here while we're refreshing... Our reload data should kickoff after both of these have completed, which feels fine